chain/store/stateGenerator.go

```diff
@@ -63,7 +63,23 @@ func (cs *ChainStore) spendTransaction(states *StateDB, txn *transaction.Transac
 		states.IncrNonce(pg[0])
 
 		registerNamePayload := pl.(*pb.RegisterName)
-		states.setName(registerNamePayload.Registrant, registerNamePayload.Name)
+		if height > config.OldNameRegistrationHeight {
+			if err = states.registerName(registerNamePayload.Name, registerNamePayload.Registrant, uint32(config.MaxNameDuration.GetValueAtHeight(height))+height); err != nil {
+				return err
+			}
+			donationAddress, err := ToScriptHash(config.DonationAddress)
+			if err != nil {
+				return err
+			}
+			if err = states.UpdateBalance(pg[0], config.NKNAssetID, config.NameRegistrationFee, Subtraction); err != nil {
```

Comment:
`NameRegistrationFee` should also be in transaction payload (all charge to signer's account should always be explicit), and in config we should change `NameRegistrationFee` to `MinNameRegistrationFee`.

Reply:
Transaction fee should also be handled.
The registration and transaction fee logic should be very similar to generate id fee, including transaction field, transaction handling, verification, etc. You can check that